File: src/main/kotlin/code/nebula/cipherquest/advisor/LoggingAdvisor.java
package code.nebula.cipherquest.advisor;

import org.slf4j.*;
import org.springframework.ai.chat.client.*;
import org.springframework.ai.chat.model.*;

import java.util.*;

public class LoggingAdvisor implements RequestResponseAdvisor {

    private static final Logger logger = LoggerFactory.getLogger(LoggingAdvisor.class);

    @Override
    public AdvisedRequest adviseRequest(AdvisedRequest request, Map<String, Object> context) {
        logger.info("Request: " + request);
        return request;
    }

    @Override
    public ChatResponse adviseResponse(ChatResponse response, Map<String, Object> context) {
		logger.info("Token: " + response.getMetadata().getUsage().getTotalTokens());

        return response;
    }
}

File: src/main/kotlin/code/nebula/cipherquest/advisor/LoggingAdvisor.kt
package code.nebula.cipherquest.advisor

import org.slf4j.Logger
import org.slf4j.LoggerFactory
import org.springframework.ai.chat.client.AdvisedRequest
import org.springframework.ai.chat.client.RequestResponseAdvisor
import org.springframework.ai.chat.model.ChatResponse

class LoggingAdvisor : RequestResponseAdvisor {
    override fun adviseRequest(
        request: AdvisedRequest,
        context: Map<String, Any>,
    ): AdvisedRequest {
        logger.info("Request: $request")
        return request
    }

    override fun adviseResponse(
        response: ChatResponse,
        context: Map<String, Any>,
    ): ChatResponse {
        logger.info("Token: " + response.metadata.usage.totalTokens)

        return response
    }

    companion object {
        private val logger: Logger = LoggerFactory.getLogger(LoggingAdvisor::class.java)
    }
}
